client/app/veeam/storage/veeam-storage.html

```diff
@@ -9,57 +9,33 @@
-                <tr class="oui-table__row">
-                    <td colspan="5" class="oui-table__cell">
-                        <oui-spinner></oui-spinner>
-                    </td>
-                </tr>
-            </tbody>
-            <tbody class="oui-table__body" data-ng-hide="VeeamStorageCtrl.storageInfos.loading">
-                <tr class="oui-table__row cui-dropdown-menu-container"
-                    data-ng-repeat="storage in VeeamStorageCtrl.storageInfos.data | filter:VeeamStorageCtrl.searchText track by $index">
-                    <td class="oui-table__cell" data-ng-bind="storage.inventoryName"></td>
-                    <td class="oui-table__cell">
-                        <span data-ng-show="storage.quotaUsed" data-ng-bind="storage.quotaUsed.value | bytes:2:false:storage.quotaUsed.unit"></span>
-                        <span data-ng-show="!storage.quotaUsed">0 B</span>
-                    </td>
-                    <td class="oui-table__cell" data-ng-bind="storage.quota.value | bytes:2:false:storage.quota.unit"></td>
-                    <td class="oui-table__cell" data-ng-bind="VeeamStorageCtrl.getRegionText(storage.replicationZone)"></td>
-                    <td class="oui-table__cell oui-table__cell_action">
-                        <oui-action-menu data-align="end" data-compact>
-                            <oui-action-menu-item data-text="{{ 'veeam_storage_update_title' | translate }}"
-                                                  data-on-click="VeeamStorageCtrl.updateQuota(storage.inventoryName)"
-                                                  data-disabled="(VeeamStorageCtrl.capabilities.data.maxQuota | bytes:0:false:'GB':true) <= (VeeamStorageCtrl.capabilities.data.defaultQuota | bytes:0:false:'GB':true)"
-                                                  data-ng-if="!VeeamStorageCtrl.capabilities.loading">
-                            </oui-action-menu-item>
-                        </oui-action-menu>
-                    </td>
-                </tr>
-            </tbody>
-        </table>
+        <oui-datagrid rows="VeeamStorageCtrl.storageInfos.data" class="datagrid-without-pagination">
+            <extra-top>
+                <button class="oui-button oui-button_secondary"
+                    data-ng-if="!VeeamStorageCtrl.actions.loading"
+                    data-translate="veeam_storage_add_storage"
+                    data-ng-click="VeeamStorageCtrl.addStorage()">
+                </button>
+                <oui-spinner data-ng-if="VeeamStorageCtrl.actions.loading"></oui-spinner>
+            </extra-top>
+            <oui-column data-title="::'veeam_table_storage' | translate" data-property="inventoryName" data-sortable="asc" data-type="string" data-searchable></oui-column>
+            <oui-column data-title="::'veeam_table_usage' | translate">
+                <span data-ng-if="$row.quotaUsed" data-ng-bind="$row.quotaUsed.value | bytes:2:false:$row.quotaUsed.unit"></span>
```

Comment:
I'm not sure this works, angular directives are not understood in `oui-datagrid` yet

Reply:
It is actually working. Have used in many places :)

Reply:
@marie-j they don't work on `oui-columns` but they work on the content 😃